org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDocumentLifeCycleHandler.java

```diff
@@ -266,9 +267,12 @@ public IStatus publishDiagnostics(IProgressMonitor monitor) throws JavaModelExce
 		if (monitor.isCanceled()) {
 			return Status.CANCEL_STATUS;
 		}
-		List<ICompilationUnit> validateCopy =
-			preferenceManager.getPreferences().isValidateAllOpenBuffersOnChanges()
-				? Arrays.asList(JavaCore.getWorkingCopies(null)) : new ArrayList<>(toValidate);
+		Set<ICompilationUnit> validateCopy = new LinkedHashSet<>(toValidate);
```

Comment:
Makes sense. "All working copies" should definitely include the ones in `toValidate`, so this would just ensure those in `toValidate` are processed first when `java.edit.validateAllOpenBuffersOnChanges` is set to `true`.

CC'ing @testforstephen in case he finds something wrong about doing this. Although, it should technically improve responsiveness from a user's perspective.

Reply:
it looks fine to me.